service/dap/server.go

```diff
@@ -1082,15 +1140,26 @@ func (s *Session) getPackageDir(pkg string) string {
 
 // newNoDebugProcess is called from onLaunchRequest (run goroutine) and
 // requires holding mu lock. It prepares process exec.Cmd to be started.
-func (s *Session) newNoDebugProcess(program string, targetArgs []string, wd string) (*exec.Cmd, error) {
+func (s *Session) newNoDebugProcess(program string, targetArgs []string, wd string) (cmd *exec.Cmd, err error) {
 	if s.noDebugProcess != nil {
 		return nil, fmt.Errorf("another launch request is in progress")
 	}
-	cmd := exec.Command(program, targetArgs...)
-	cmd.Stdout, cmd.Stderr, cmd.Stdin, cmd.Dir = os.Stdout, os.Stderr, os.Stdin, wd
-	if err := cmd.Start(); err != nil {
+
+	cmd = exec.Command(program, targetArgs...)
```

Comment:
It's wrong to unconditionally redirect stdout and stderr because programs can behave differently when their output is not a tty. It should only be done if outputMode is "remote".

Reply:
I don't understand... 
When outputMode is not "remote", I will rewrite the output to os.stderr and os.stdout.
Can you give me an error case?

Reply:
https://github.com/mattn/go-isatty or https://github.com/pkg/term/

Reply:
I still can't understand...
Will there be a difference between "cmd.stdout->os.stdout" and "cmd.stdout->os.pipe->os.stdout" on non-tty terminal?
Is it possible to call os.Pipe() on a non-tty terminal and get an error?
I'm sorry, I don't know much about non-tty terminal.

Reply:
>  Will there be a difference between "cmd.stdout->os.stdout" and "cmd.stdout->os.pipe->os.stdout" on non-tty terminal?

All terminals are tty terminals.

>  Is it possible to call os.Pipe() on a non-tty terminal and get an error?

No.

>  I'm sorry, I don't know much about non-tty terminal.

The problem is with tty terminals.

Reply:
Okay, I got it.
done.